app/src/main/java/org/thoughtcrime/securesms/loki/protocol/ClosedGroupsProtocolV2.kt

```diff
@@ -359,6 +351,10 @@ object ClosedGroupsProtocolV2 {
         }
         // Generate the new encryption key pair
         val newKeyPair = Curve.generateKeyPair()
+        do {
+            // make sure we set the pendingKeyPair or wait until it is not null
+        } while (!pendingKeyPair.compareAndSet(null, newKeyPair))
```

Comment:
`pendingKeyPair` has to be group specific. Otherwise if you're in multiple groups, responding to multiple of these messages, you can send the key pair of one group to another group.

Reply:
oh yeah of course :upside_down_face: